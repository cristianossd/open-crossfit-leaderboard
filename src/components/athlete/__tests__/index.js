import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import Athlete from '../';

Object.defineProperty(window, 'innerWidth', { writable: true });

const attrsMock = {
  overallRank: 1,
  overallScore: 1,
  competitorName: 'competitor mock',
  affiliateName: 'affiliate mock',
  scores: [
    { rank: 1, scoreDisplay: '299 reps' },
    { rank: 2, scoreDisplay: '06:32' },
    { rank: 3, scoreDisplay: '95 reps' },
    { rank: 4, scoreDisplay: '100 reps' },
    { rank: 5, scoreDisplay: '15:10' },
  ],
};

describe('<Athlete />', () => {
  it('render correctly', () => {
    window.innerWidth = '1800';
    const wrapper = shallow(<Athlete attrs={attrsMock} index={1} />);

    expect(wrapper.find('.Athlete--sm').length).toBe(0);
    expect(wrapper.find('.Athlete--md').length).toBe(0);
    expect(wrapper.find('.Athlete--lg').length).toBe(1);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('render in medium screens', () => {
    window.innerWidth = '1320';
    const wrapper = shallow(<Athlete attrs={attrsMock} index={1} />);

    expect(wrapper.find('.Athlete--sm').length).toBe(0);
    expect(wrapper.find('.Athlete--md').length).toBe(1);
    expect(wrapper.find('.Athlete--lg').length).toBe(0);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('render in small screens', () => {
    window.innerWidth = '768';
    const wrapper = shallow(<Athlete attrs={attrsMock} index={1} />);

    expect(wrapper.find('.Athlete--sm').length).toBe(1);
    expect(wrapper.find('.Athlete--md').length).toBe(0);
    expect(wrapper.find('.Athlete--lg').length).toBe(0);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
