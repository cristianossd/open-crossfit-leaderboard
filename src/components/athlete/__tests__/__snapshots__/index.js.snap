// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Athlete /> render correctly 1`] = `
<tr
  className="Athlete--lg"
  key="1"
>
  <th
    scope="row"
  >
    1
  </th>
  <td>
    <strong>
      competitor mock
    </strong>
    <br />
    <small>
      affiliate mock
    </small>
  </td>
  <td>
    <strong>
      1
    </strong>
  </td>
  <td
    key="0"
  >
    <strong>
      1st
    </strong>
    <br />
    <small>
      (
      299 reps
      )
    </small>
  </td>
  <td
    key="1"
  >
    <strong>
      2nd
    </strong>
    <br />
    <small>
      (
      06:32
      )
    </small>
  </td>
  <td
    key="2"
  >
    <strong>
      3rd
    </strong>
    <br />
    <small>
      (
      95 reps
      )
    </small>
  </td>
  <td
    key="3"
  >
    <strong>
      4th
    </strong>
    <br />
    <small>
      (
      100 reps
      )
    </small>
  </td>
  <td
    key="4"
  >
    <strong>
      5th
    </strong>
    <br />
    <small>
      (
      15:10
      )
    </small>
  </td>
</tr>
`;

exports[`<Athlete /> render in medium screens 1`] = `
<tr
  className="Athlete--md"
  key="1"
>
  <td>
    <h6>
      <strong>
        1
         (
        1
        )
         
        competitor mock
      </strong>
      <br />
         
      <small>
        affiliate mock
      </small>
    </h6>
  </td>
  <td
    key="0"
  >
    <strong>
      1st
    </strong>
    <br />
    <small>
      (
      299 reps
      )
    </small>
  </td>
  <td
    key="1"
  >
    <strong>
      2nd
    </strong>
    <br />
    <small>
      (
      06:32
      )
    </small>
  </td>
  <td
    key="2"
  >
    <strong>
      3rd
    </strong>
    <br />
    <small>
      (
      95 reps
      )
    </small>
  </td>
  <td
    key="3"
  >
    <strong>
      4th
    </strong>
    <br />
    <small>
      (
      100 reps
      )
    </small>
  </td>
  <td
    key="4"
  >
    <strong>
      5th
    </strong>
    <br />
    <small>
      (
      15:10
      )
    </small>
  </td>
</tr>
`;

exports[`<Athlete /> render in small screens 1`] = `
<tr
  className="Athlete--sm"
  key="1"
>
  <td>
    <Accordion
      accordion={false}
      className="accordion"
      onChange={[Function]}
    >
      <inject-AccordionItem-with-accordionStore>
        <inject-AccordionItemTitle-with-accordionStore-uuid>
          <h6
            className="u-position-relative"
          >
            <strong>
              1
               (
              1
              )
              competitor mock
            </strong>
            <br />
                 
            <small>
              affiliate mock
            </small>
            <div
              className="accordion__arrow"
              role="presentation"
            />
          </h6>
        </inject-AccordionItemTitle-with-accordionStore-uuid>
        <inject-AccordionItemBody-with-accordionStore-uuid>
          <ul>
            <li
              key="18.1:"
            >
              <strong>
                 
                18.1:
              </strong>
               
              1st
               (
              299 reps
              )
            </li>
            <li
              key="18.2:"
            >
              <strong>
                 
                18.2:
              </strong>
               
              2nd
               (
              06:32
              )
            </li>
            <li
              key="18.2a:"
            >
              <strong>
                 
                18.2a:
              </strong>
               
              3rd
               (
              95 reps
              )
            </li>
            <li
              key="18.3:"
            >
              <strong>
                 
                18.3:
              </strong>
               
              4th
               (
              100 reps
              )
            </li>
            <li
              key="18.4:"
            >
              <strong>
                 
                18.4:
              </strong>
               
              5th
               (
              15:10
              )
            </li>
          </ul>
        </inject-AccordionItemBody-with-accordionStore-uuid>
      </inject-AccordionItem-with-accordionStore>
    </Accordion>
  </td>
</tr>
`;
